hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/producers/formats/v1/BlockStreamFormatV1.java

```diff
@@ -69,8 +111,32 @@ public Bytes serializeEventMetadata(@NonNull final EventMetadata eventMetadata)
 
     /** {@inheritDoc} */
     @Override
-    public Bytes serializeSystemTransaction(@NonNull final SystemTransaction systemTransaction) {
-        return null;
+    public Bytes serializeSystemTransaction(@NonNull final ConsensusTransaction systemTransaction) {
+        var systemTxnBuilder = SystemTransaction.newBuilder();
+
+        switch (systemTransaction) {
+            case StateSignatureTransaction stateSignatureTransaction -> {
```

Comment:
This code seems to suggest the platform will gather the signatures and generate a single `StateSignatureTransaction`. I was under the impression that signatures for each node come in different system transactions; do we not need to collect those somewhere?